lib/src/main/java/graphql/nadel/validation/hydration/NadelHydrationVirtualTypeValidation.kt

```diff
@@ -0,0 +1,120 @@
+package graphql.nadel.validation.hydration
+
+import graphql.nadel.definition.hydration.isHydrated
+import graphql.nadel.definition.virtualType.isVirtualType
+import graphql.nadel.engine.blueprint.NadelVirtualTypeContext
+import graphql.nadel.engine.util.unwrapAll
+import graphql.nadel.validation.NadelValidationContext
+import graphql.nadel.validation.NadelValidationInterimResult
+import graphql.nadel.validation.NadelValidationInterimResult.Success.Companion.asInterimSuccess
+import graphql.nadel.validation.onError
+import graphql.nadel.validation.onErrorCast
+import graphql.schema.GraphQLFieldDefinition
+import graphql.schema.GraphQLFieldsContainer
+import graphql.schema.GraphQLNamedType
+
+private data class NadelVirtualTypeMapping(
+    val virtualType: GraphQLNamedType,
+    val backingType: GraphQLNamedType,
+)
+
+private class NadelHydrationVirtualTypeValidationContext {
+    private val visited = mutableSetOf<Pair<String, String>>()
+
+    /**
+     * @return true to continue
+     */
+    fun visit(mapping: NadelVirtualTypeMapping): Boolean {
+        val reference = mapping.virtualType.name to mapping.backingType.name
+        return visited.add(reference)
+    }
+}
+
+/**
+ * This only generates a [NadelVirtualTypeContext] assuming a valid virtual type.
+ *
+ * A virtual types are validated in [graphql.nadel.validation.NadelVirtualTypeValidation]
```

Comment:
Refer to this comment.

This class does a similar job to `NadelVirtualTypeValidation` but assumes a valid virtual type.

This class just gathers the type mapping together for hydration.